.github/workflows/main.yml

```diff
@@ -0,0 +1,55 @@
+name: CI/CD
+
+on: [push]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Cache Bitcoin node
+        id: cache
+        uses: actions/cache@v2
+        with:
+          path: ./bitcoin-0.20.0
+          key: ${{ runner.os }}-cache
+
+      - name: Download Bitcoin node
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: |
+          wget https://bitcoin.org/bin/bitcoin-core-0.20.0/bitcoin-0.20.0-x86_64-linux-gnu.tar.gz
+          tar -xvf bitcoin-0.20.0-x86_64-linux-gnu.tar.gz
+
+      - name: Start Bitcoin node
+        run: |
+          cp bitcoin-0.20.0/bin/* /usr/local/bin/
+          bitcoind -reindex -txindex -regtest -daemon -rpcport=18443 -rpcuser=bitcoin -rpcpassword=local321 -deprecatedrpc=signrawtransaction -fallbackfee=0.00001 -addresstype=p2sh-segwit -zmqpubrawblock=tcp://127.0.0.1:29000 -zmqpubrawtx=tcp://127.0.0.1:29000
+          sleep 5
+          bitcoin-cli -regtest -rpcport=18443 -rpcuser=bitcoin -rpcpassword=local321 generatetoaddress 101 $(bitcoin-cli -regtest -rpcport=18443 -rpcuser=bitcoin -rpcpassword=local321 getnewaddress '' 'bech32')
+
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 14.x
+
+      - name: restore lerna node_modules from cache
+        uses: actions/cache@v2
+        with:
+          key: ${{ runner.os }}-${{ hashFiles('yarn.lock') }}
+          path: |
+            node_modules
+            */*/node_modules
+
+      - name: yarn install
+        if: steps.${{ runner.os }}-${{ hashFiles('yarn.lock') }}.outputs.cache-hit != 'true'
+        run: yarn install --frozen-lockfile
+
+      - run: yarn bootstrap
+
+      - name: Build cfd-dlc-js
```

Comment:
Is this action cached? If so, awesome, don't want to rebuild cfd-dlc-js every time 😂

Reply:
Yeah good question, I cache the entire `node_modules` so if `node_modules` is present, then it skips the step entirely. If it isn't present (or if `.lock` file has changed) then it installs packages and it has to rebuild cfd-dlc-js.